### PR TITLE
feat(listing): add alerts to listing access

### PR DESCRIPTION
**Pull request by flesher1**
Adds alert for users who access deleted listings
Adds Alerts for users who access private listings
Adds alerts for users who access disabled listings
Adds alerts for users who access non-published listings
Adds alerts for users who access listings that do not exist
Fix for user access check

Closes #AMLOS-789, #AMLOS-811

How to test.
Log in as Bigbrother
log in as johnson

delete a listing, 
Make a listing private,
disable a listing,
create a draft listing, as well as awaiting approval

check access as both users to each listing
Finally, check if a user fills in their own listing id (ex: 999999)

check all warning/error alerts and if the user can view the data held in the listing.

### CODE REVIEW

File: src/components/listing/Listing.js [hunk 1]
```diff
@@ -1,27 +1,117 @@
...
+            closeFunction: () => {this.props.dispatch(goBack())},
+        };
+
+        if(listing === undefined){
+            if(!this.state.alertOpen) this.openAlert();
+        }else if(listing.approvalStatus === 'DELETED'){
+            if(!this.state.alertOpen) this.openAlert();
+            alert = {
+                ...alert,
+                title: "Sorry, looks like this listing has been deleted",
```
Comment (blynch11p) [inline src/components/listing/Listing.js]: Minor wording change request:
`Sorry, it looks like this listing...`
or
`Sorry, it appears that this listing...`

**Comment by blynch11p** [comment]
I don't know if it's fixable, or a quick fix, but, if you click the navigation tabs on the left, while an admin (or user that can access disabled listings) the alert keep popping up.

**Comment by flesher1** [comment]
@blynch11p Going to leave it as is for now. 

Currently the method will show an alert when the user makes an update to the listing in some way (review, etc.). It would be possible to show the alert once, but it would limit this use case.

**Comment by blynch11p** [comment]
Then :+1:

**Comment by irtiza-rizvi** [comment]
:+1: works as intended

**Pull request merged**